### PR TITLE
ref(breadcrumbs): Use SearchBarActionV2

### PR DESCRIPTION
**Pull request by vuluongj20**
Refactor `SearchBarActions` (which is used by `Breadcrumbs` in issues details) to use `CompactSelect`.

**Before:**
image

**After:**
image

This is part of a group PRs that will be merged together:
 - https://github.com/getsentry/sentry/pull/35226
 - https://github.com/getsentry/sentry/pull/35227
 - https://github.com/getsentry/sentry/pull/35228

Once all of these PRs are merged, I will delete v1 of `SearchBarAction` and remove the v2 tag from the newer one.

### CODE REVIEW

File: static/app/components/events/interfaces/searchBarActionV2.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  onChange: (value: string) => void;
+  placeholder: string;
+  query: string;
+  className?: string;
+  filterOptions?: FilterOption[];
+  filterSelections?: FilterOption[];
+  onFilterChange?: (options: FilterOption[]) => void;
+};
+
+function SearchBarAction({
```
Comment (evanpurkhiser) [inline static/app/components/events/interfaces/searchBarActionV2.tsx:21]: How many other places is the v1 used in?

Comment (vuluongj20) [reply static/app/components/events/interfaces/searchBarActionV2.tsx:21]: two more

Comment (vuluongj20) [reply static/app/components/events/interfaces/searchBarActionV2.tsx:21]: creating the PRs for them now

File: static/app/components/events/interfaces/breadcrumbs/index.tsx [hunk 2]
```diff
@@ -207,17 +208,17 @@ function BreadcrumbsContainer({
...
+        .filter(option => option.value.split('-')[0] === 'type')
+        .map(option => option.value.split('-')[1])
     );
 
     const checkedLevelOptions = new Set(
-      Object.values(newfilterOptions)[1]
-        .filter(filterOption => filterOption.isChecked)
-        .map(option => option.id)
+      selectedFilterOptions
+        .filter(option => option.value.split('-')[0] === 'level')
```
Comment (evanpurkhiser) [inline static/app/components/events/interfaces/breadcrumbs/index.tsx:220]: why not
```suggestion
        .filter(option => option.value.startsWith('level-'))
```

**Review by evanpurkhiser — APPROVED** [review]

File: tests/js/spec/components/events/interfaces/breadcrumbs/filter.spec.tsx [hunk 3]
```diff
@@ -1,158 +1,184 @@
...
+        onFilterChange={handleFilter}
+        onChange={() => null}
+        placeholder=""
+        query=""
+      />
     );
 
-    const filterDropdownMenu = wrapper.find('StyledContent');
+    const filterDropdownMenu = screen.getByText('Filter By');
+    userEvent.click(filterDropdownMenu);
```
Comment (priscilawebdev) [inline tests/js/spec/components/events/interfaces/breadcrumbs/filter.spec.tsx:84]:
```suggestion
    userEvent.click(screen.getByText('Filter By'));
```

File: tests/js/spec/components/events/interfaces/breadcrumbs/filter.spec.tsx [hunk 4]
```diff
@@ -1,158 +1,184 @@
...
```
Comment (priscilawebdev) [inline tests/js/spec/components/events/interfaces/breadcrumbs/filter.spec.tsx:131]:
```suggestion
    userEvent.click(screen.getByText('Filter By'));
```

File: tests/js/spec/components/events/interfaces/breadcrumbs/filter.spec.tsx [hunk 5]
```diff
@@ -1,158 +1,184 @@
...
-    expect(handleFilter).toHaveBeenCalledTimes(1);
+    expect(screen.getByText('HTTP request')).toBeInTheDocument();
+    expect(screen.getByText('Transaction')).toBeInTheDocument();
+    expect(screen.getByText('User Action')).toBeInTheDocument();
+    expect(screen.getByText('Navigation')).toBeInTheDocument();
+    expect(screen.getByText('Debug')).toBeInTheDocument();
+    expect(screen.getAllByText('Error')[0]).toBeInTheDocument();
+
+    const httpRequestItem = screen.getByText('HTTP request');
+    userEvent.click(httpRequestItem);
```
Comment (priscilawebdev) [inline tests/js/spec/components/events/interfaces/breadcrumbs/filter.spec.tsx:146]:
```suggestion
    userEvent.click(screen.getByText('HTTP request'));
```

File: tests/js/spec/components/events/interfaces/breadcrumbs/filter.spec.tsx [hunk 6]
```diff
@@ -1,158 +1,184 @@
...
+        onFilterChange={handleFilter}
+        onChange={() => null}
+        placeholder=""
+        query=""
+      />
     );
 
-    const filterDropdownMenu = wrapper.find('StyledContent');
+    const filterDropdownMenu = screen.getByText('Filter By');
+    userEvent.click(filterDropdownMenu);
```
Comment (priscilawebdev) [inline tests/js/spec/components/events/interfaces/breadcrumbs/filter.spec.tsx:167]:
```suggestion
    userEvent.click(screen.getByText('Filter By'));
```

File: tests/js/spec/components/events/interfaces/breadcrumbs/filter.spec.tsx [hunk 7]
```diff
@@ -1,158 +1,184 @@
...
-    expect(firstItem.text()).toBe('Info');
+    expect(screen.getByText('info')).toBeInTheDocument();
+    expect(screen.getByText('error')).toBeInTheDocument();
 
     // Check Item
-    expect(firstItem.find('CheckboxFancy').props().isChecked).toBeTruthy();
-
-    firstItem.simulate('click');
+    const infoItem = screen.getByText('info');
+    userEvent.click(infoItem);
```
Comment (priscilawebdev) [inline tests/js/spec/components/events/interfaces/breadcrumbs/filter.spec.tsx:179]:
```suggestion
    userEvent.click(screen.getByText('info'));
```

**Comment by priscilawebdev** [comment]
I've just tested it locally and the UI doesn't seem to been shown all the Error Badge:

image

**Pull request merged**